integration/osio_provider_test.go

```diff
@@ -0,0 +1,117 @@
+package integration
+
+import (
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/containous/traefik/integration/try"
+	"github.com/containous/traefik/log"
+	"github.com/go-check/check"
+	checker "github.com/vdemeester/shakers"
+)
+
+// OSIOProviderSuite
+type OSIOProviderSuite struct{ BaseSuite }
+
+func (s *OSIOProviderSuite) TestOSIOProvider(c *check.C) {
+	// configure OSIO
+	os.Setenv("WIT_URL", witURL)
+	os.Setenv("AUTH_URL", authURL)
+	os.Setenv("SERVICE_ACCOUNT_ID", "any-id")
+	os.Setenv("SERVICE_ACCOUNT_SECRET", "anysecret")
+	witServer := startOSIOServer(9090, serveWITRequest)
+	defer witServer.Close()
+	authServer := startOSIOServer(9091, serverAUTHRequest2)
+	defer authServer.Close()
+
+	// Start Traefik
+	cmd, display := s.traefikCmd(withConfigFile("fixtures/osio_provider_config.toml"))
+	defer display(c)
+	err := cmd.Start()
+	c.Assert(err, checker.IsNil)
+	defer cmd.Process.Kill()
+
+	// Start OSIO servers
+	ts1 := startOSIOServer(8081, nil)
+	defer ts1.Close()
+	ts2 := startOSIOServer(8082, nil)
+	defer ts2.Close()
+
+	// make multiple reqeust on some time gap
+	// note, req has 'Bearer 2222' so it should go to 'http://127.0.0.1:8082' check serverAUTHRequest2()
+	// check serverAUTHRequest2(), return 'http://127.0.0.1:8082' cluster for only first time
+	// so first few response would be 'HTTP 200 OK' and then rest would be 'HTTP 404 not found'
```

Comment:
Check comments in test case.  This test cover end-to-end OSIO backend functionality.